File: lib/aerox/scene/components/CameraComponent.cpp
#include <aerox/scene/components/CameraComponent.hpp>
#include "aerox/scene/Scene.hpp"
#include "aerox/scene/objects/SceneObject.hpp"
//#include <glm/gtx/transform.hpp>
#include <glm/ext/matrix_transform.hpp>

namespace aerox::scene {

glm::mat4 CameraComponent::GetViewMatrix() const {
  const auto transform = GetWorldTransform();
  return inverse(transform.GetLocationMatrix() * transform.GetRotationMatrix());
}

glm::mat4 CameraComponent::GetProjection(const float aspectRatio) const {
  return glm::rotate(glm::perspective(glm::radians(fieldOfView), aspectRatio,  nearClipPlane,farClipPlane),glm::radians(180.f),{1,0,0});
}

void CameraComponent::Draw(
    drawing::SceneFrameData *frameData, const math::Transform &parentTransform) {
  
}

void CameraComponent::SetRelativeTransform(const math::Transform &val) {
  RenderedComponent::SetRelativeTransform(val);
  
}
}

File: aerox.Runtime.Scene/Components/CameraComponent.cs
using aerox.Runtime.Math;

namespace aerox.Runtime.Scene.Components;

public class CameraComponent : RenderedComponent
{


    public float FieldOfView = 90.0f;

    public float NearClipPlane = 1f;

    public float FarClipPlane = 1000.0f;
    public Matrix4 GetViewMatrix()
    {
        var worldTransform = GetWorldTransform();
        Matrix4 transformMatrix = new Transform()
        {
            Location = worldTransform.Location,
            Rotation = worldTransform.Rotation
        };

        return transformMatrix.Inverse();
    }


    public Matrix4 GetProjection(float width, float height)
    {
        return Glm.Perspective(float.DegreesToRadians(FieldOfView), width / height, NearClipPlane, FarClipPlane).RotateDeg(180.0f,new Vector3<float>(1f,0f,0f));
    }
}